File: jsp_web/src/main/webapp/js/api/api1.js
console.log( 'api1.js 실행' );

// * AJAX 이용한 데이터포털의 데이터 가져오기

// 공공데이터 : 시도별 미세먼지 실시간 현황
getapi2();
function getapi2(){
	
	$.ajax({
		url : "https://apis.data.go.kr/B552584/ArpltnInforInqireSvc/getCtprvnRltmMesureDnsty?serviceKey=3zQogYLFxx%2BFClPZRnCo%2FD0rGd%2BCXtMLZ5%2Fw756iUPnyOA8xKUXrzRs09ZtYjAWsTe3%2Fu5aAi4%2BZX0%2FZoycn3A%3D%3D&returnType=json&numOfRows=200&pageNo=1&sidoName=%EA%B2%BD%EA%B8%B0&ver=1.0",
		method : "get",
		success : ( r ) => {
			console.log( r );
		}
	})
}


// 공공데이터 : 전기차충전소 현황
getapi1();
function getapi1(){
	$.ajax({
		url : "https://api.odcloud.kr/api/15090398/v1/uddi:6fe0e3f2-0285-4999-9edf-995afe19a6ea?page=1&perPage=96&serviceKey=3zQogYLFxx%2BFClPZRnCo%2FD0rGd%2BCXtMLZ5%2Fw756iUPnyOA8xKUXrzRs09ZtYjAWsTe3%2Fu5aAi4%2BZX0%2FZoycn3A%3D%3D",
		method : "get",
		success : ( r ) => {
			console.log( r );
			document.querySelector('.totalcount').innerHTML = `전기차 충전소 개수 : ${ r.totalCount }`;
			
			let html = `<tr>
							<th> 시군명 </th>
							<th> 소재지우편번호 </th>
							<th> 소재지도로명주소 </th>
							<th> 소재지지번주소 </th>
							<th> 운영기관명 </th>
							<th> 충전기타입명 </th>
							<th> 충전소명 </th>
							<th> 경도(WGS84)</th>
							<th> 위도(WGS84) </th>
						</tr>`
			
			r.data.forEach( (o) =>{
				// 객체내 필드 호출하는 방법 : 1. 객체명.필드명 		2. 객체명['필드명']
				
				html += `<tr>
							<td> ${ o.시군명 } </td>
							<td> ${ o.소재지우편번호 } </td>
							<td> ${ o.소재지도로명주소 } </td>
							<td> ${ o.소재지지번주소 } </td>
							<td> ${ o.운영기관명 } </td>
							<td> ${ o.충전기타입명 } </td>
							<td> ${ o.충전소명 } </td>
							<td> ${ o['경도(WGS84)'] }</td>
							<td> ${ o['위도(WGS84)'] } </td>
						</tr>`
			})
			
			document.querySelector('.table').innerHTML = html ;
		}
	})
}


/*

경도(WGS84) : "126.8522331753"
소재지도로명주소 : "경기도 안산시 상록구 선진로 92"
소재지우편번호 : 15636
소재지지번주소 : "경기도 안산시 상록구 사동 1502번지"
시군명 : "안산시"
운영기관명 : "한국전력"
위도(WGS84) : "37.2787468594"
충전기타입명 : "DC콤보"
충전소명 :

*/

File: jsp_web/src/main/java/controller/api/Api1.java
package controller.api;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.URL;

import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;


@WebServlet("/api1")
public class Api1 extends HttpServlet {
	private static final long serialVersionUID = 1L;
       
    
    public Api1() { super(); }
    
	
	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
		
		// 1. 공공데이터 포털에서 신청한 데이터 가져오기 [ 안산시 전기차충전소 현황 ]
			// URL 클래스
			// 1. .openStream() : 해당 객체의 연결된 url 스트림 제공 함수 [ 반환 : InputStream ] 
		// URL url = new URL("URL주소");
		URL url = new URL("https://api.odcloud.kr/api/15090398/v1/uddi:6fe0e3f2-0285-4999-9edf-995afe19a6ea?page=1&perPage=96&serviceKey=3zQogYLFxx%2BFClPZRnCo%2FD0rGd%2BCXtMLZ5%2Fw756iUPnyOA8xKUXrzRs09ZtYjAWsTe3%2Fu5aAi4%2BZX0%2FZoycn3A%3D%3D");
		
		// 2. 해당 URL의 데이터 [ 스트림(바이트) ] 읽어오기 위한 스트림 객체 생성
		// InputStream inputStream = url.openStream();
		// InputStreamReader reader = new InputStreamReader( url.openStream() , "UTF-8" );
		
		// *
		BufferedReader bf = new BufferedReader( new InputStreamReader( url.openStream() , "UTF-8") );
		
		
			// byte[] array1 = new byte[ 100000 ];
			// char[] array2 = new char[ 100000 ];
		
		// 3. 스트림으로 바이트 읽어오기  
		// inputStream.read( array1 ); 	// 바이트 배열로 읽어오기 
		// reader.read( array2 );			// 문자 배열로 읽어오기
		
		// * 
		String result = bf.readLine();					// 모든 바이트를 읽은 후 문자열(String) 반환
		
		System.out.println( result );
		
	}

	
	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
	
	}

}
